### DISCUSSION TITLE
Geofence is top, but it would be helpful and a great thing if....

### DISCUSSION THREAD

**Discussion by dansche** [teslamate-org/teslamate#1950] [Ideas]
First of all, thanks for developing this, absolutely in love with it. Great tool!!!!

One thing would be great, usually I load from my PV environment during Sunshine... which is during the day usually ;-) The cost I calculate for this is the price I receive from my Power Provider to sell my energy if I have to much production.

As I do load sometimes as well during night time I have to buy on the same spot the energy to another price, so it would be great, if I can add another pricing let's say and enter a time the pricing is valid.

So let's say GeoFence
9 - 5 PM price X 
5 - 9 PM/AM price Y

This would give a more exact calculation and I would be able to automate my loadcycle.

My Wallbox is able to see if my car has 80% power, if not, it starts automatically at 2 AM to load with 11KW, if there is 80% loaded, the Wallbox will not start to charge.

Don't hesitate to ask for more information if needed.

Thanks

**Comment by Bz16** [comment]
More utilities are adopting time-of-use billing models, so this should be really helpful for tracking at-home charging costs. Hack, even the Tesla app has rudimentary charging price estimates that supports TOU pricing.
